File: client/src/app/_models/admin/orderAssessmentItem.ts
import { IOrderAssessmentItemQ } from "./orderAssessmentItemQ";


export interface IOrderAssessmentItem
{
     id: number;
     orderAssessmentId: number;
     assessmentRef: string;
     orderItemId: number;
     customerName: string;
     professionId: number;
     professionName: string;
     orderId: number;
     orderNo: number;
     approvedBy: string;
     dateDesigned: Date;     
     requireCandidateAssessment: boolean;
     designedBy: string;
     orderAssessmentItemQs: IOrderAssessmentItemQ[];
}

export class OrderAssessmentItem implements IOrderAssessmentItem
{
     id= 0;
     orderAssessmentId= 0;
     assessmentRef= '';
     orderItemId= 0;
     customerName= '';
     professionId= 0;
     professionName= '';
     orderId= 0;
     orderNo= 0;
     approvedBy= '';
     dateDesigned = new Date;     
     requireCandidateAssessment = false;
     designedBy= '';
     orderAssessmentItemQs: IOrderAssessmentItemQ[] = [];
}

File: api/Entities/Admin/Order/OrderAssessmentItem.cs
namespace api.Entities.Admin.Order
{
    public class OrderAssessmentItem: BaseEntity
    {
        public int OrderAssessmentId { get; set; }
        public string AssessmentRef { get; set; }
        public int OrderItemId {get; set;}
        public string ProfessionGroup {get; set;}
        public string CustomerName { get; set; }
        public string ProfessionName { get; set; }
        public int ProfessionId { get; set; }
        public int OrderNo { get; set; }
        public int OrderId { get; set; }
        public string ApprovedBy { get; set; }
        public DateTime DateDesigned { get; set; }
        public bool RequireCandidateAssessment { get; set; }
        public string DesignedBy { get; set; }
        public ICollection<OrderAssessmentItemQ> OrderAssessmentItemQs { get; set; }
    }
}